src/hotspot/share/runtime/objectMonitor.hpp

```diff
@@ -202,12 +202,18 @@ class ObjectMonitor : public CHeapObj<mtObjectMonitor> {
   static OopHandle& vthread_cxq_head() { return _vthread_cxq_head; }
   static ParkEvent* vthread_unparker_ParkEvent() { return _vthread_unparker_ParkEvent; }
 
+  #define OM_PERFDATA_OP(f, op_str)                 \
+    do {                                            \
+      if (ObjectMonitor::_sync_ ## f != nullptr) {  \
+        ObjectMonitor::_sync_ ## f->op_str;         \
+      }                                             \
+    } while (0)
+
   // Only perform a PerfData operation if the PerfData object has been
   // allocated and if the PerfDataManager has not freed the PerfData
   // objects which can happen at normal VM shutdown. Additionally, we
```

Comment:
This commentary also applies to the OM_PERFDATA_OP case. But we can also say to use that form when not safepoint-safe, and the below when safepoint-safe.

Reply:
Right. See if got a good level of guidance in the comments now?